### PR TITLE
feat(next, vital-demo-next, __vital-next__, __minimal-next__): Add support to run a NextJS site

### PR DESCRIPTION
**Pull request by marcopagliarulo**
## Changes

Added the package @bodiless/next which provides the integration between Bodiless and Nextjs 13
Added Minimal and Vital starter based on Next, and Vital Demo based on next

### CODE REVIEW

File: sites/__vital-next__/package.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  "author": "Chris Oden <coden@its.jnj.com>",
+  "homepage": "",
+  "license": "Apache-2.0",
+  "private": true,
+  "directories": {
+    "lib": "lib",
+    "test": "__tests__"
+  },
+  "dependencies": {
+    "--vital--": "^1.0.0-rc.37",
```
Comment (wodenx) [inline sites/__vital-next__/package.json:14]: Why are we depending on --vital--?  Isn't this package intened to replace --vital-- for nextjs apps?

File: sites/__vital-next__/public/files/pages/styleguide/list/93fa4c213a3b08eaff1795428dad3c15/test.pdf
Comment (wodenx) [inline sites/__vital-next__/public/files/pages/styleguide/list/93fa4c213a3b08eaff1795428dad3c15/test.pdf:1]: Shouldn't public directory be gitignored?

File: sites/__vital-next__/src/data/site/Footer$copyright.json
Comment (wodenx) [inline sites/__vital-next__/src/data/site/Footer$copyright.json:1]: Not sure we should have any ocntent here

File: sites/__vital-next__/src/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import type { AppProps } from 'next/app';
+import '../css/style.css';
+// Add DM Sans font from --vital--
```
Comment (wodenx) [inline sites/__vital-next__/src/pages/_app.tsx:4]: Should not be importing from --vital--

File: packages/bodiless-next/src/Webpack/Config.ts [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+import type { BodilessNextConfig } from '../NextConfig/bodilessNextConfigLoader';
+
+type BodilessNextConfigWithNext = BodilessNextConfig & {
+  nextWebpack: any
+};
+
+const REGEXP = /\.gatsby/;
+const REPLACEMENT = '.next';
+
+const createLogger = (log = true) => (message :String) => {
```
Comment (dewen) [inline packages/bodiless-next/src/Webpack/Config.ts:32]: `(message :String)` => `(message: string)`

File: packages/bodiless-next/src/Webpack/Config.ts [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+  // eslint-disable-next-line no-console
```
Comment (dewen) [inline packages/bodiless-next/src/Webpack/Config.ts:33]: do we need to hide TS warning by line?

File: tsconfig.settings.json [hunk 5]
```diff
@@ -58,6 +58,8 @@
       "@bodiless/layouts-ui/*": ["bodiless-layouts-ui/src/*"],
       "@bodiless/navigation": ["bodiless-navigation/src"],
       "@bodiless/navigation/*": ["bodiless-navigation/src/*"],
+      "@bodiless/next": ["bodiless-oidc/next"],
```
Comment (dewen) [inline tsconfig.settings.json:61]: `bodiless-next/src`

File: sites/vital-demo-next/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
```
Comment (dewen) [inline sites/vital-demo-next/tsconfig.json:4]: "src/**/*.ts"?

File: sites/vital-demo-next/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    "**/*.tsx"
+  ],
+  "exclude": [
+    "node_modules"
```
Comment (dewen) [inline sites/vital-demo-next/tsconfig.json:8]: exclude `lib` as well?

File: sites/__vital-next__/public/files/pages/styleguide/list/93fa4c213a3b08eaff1795428dad3c15/test.pdf
Comment (marcopagliarulo) [reply sites/__vital-next__/public/files/pages/styleguide/list/93fa4c213a3b08eaff1795428dad3c15/test.pdf:1]: public for nextJS is what static is for Gatsby, any static asset should go there

File: sites/vital-demo-next/tsconfig.json (hunk 7 above)
Comment (marcopagliarulo) [reply sites/vital-demo-next/tsconfig.json:8]: site transpilatation doesn't output to lib

File: packages/__vital_next__/src/shadow/@bodiless/vital-layout/Header.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    ...vitalLink.Default,
+    // Make the link not editable.
+    Schema: {},
+  }),
+  asLanguageSelector
+);
+
+const Default = asHeaderToken(
+  vitalHeaderBase.Default,
+  // @todo Restore this if you don't want search on your site.
```
Comment (dewen) [inline packages/__vital_next__/src/shadow/@bodiless/vital-layout/Header.ts:34]: `if you want search on your site.`

**Review by dewen — APPROVED** [review]

**Pull request merged**